### PR TITLE
Add party processing in responsibility

### PR DESCRIPTION
**Pull request by rshewitt**
related to [#4639](https://github.com/GSA/data.gov/issues/4639)

## Modules
- add party processing to responsibility module
  - create party module and test
    - add organization module and test
      - add contact module and test
        - add address module and test
        - add phone module and test
        - add online resource module and test
      - add browse graphic module and test
    - add individual module and test

## General Improvements
- update readme with iso19115-3 work
- update some xpaths to look for the correct elements
- cleanup comments and debug import
- add more content to fixture

### CODE REVIEW

File: lib/adiwg/mdtranslator/readers/iso19115_3/modules/module_geographic_extent.rb [hunk 1]
```diff
@@ -52,17 +55,18 @@ def self.process_geo_extent(xGeoElem, hGeoExt, hResponseObj)
...
-                  hGeoExt = intMetadataClass.newGeographicExtent
+                  geoExtents = []
 
                   xGeoElems = xExtent.xpath(@@geoElemXPath)
                   xGeoElems.each do |xgeoelem|
-                     process_geo_extent(xgeoelem, hGeoExt, hResponseObj)
+                     geoExtents << process_geo_extent(xgeoelem, hResponseObj)
                   end
-                  hGeoExt
+                  geoExtents
```
Comment (jbrown-xentity) [inline lib/adiwg/mdtranslator/readers/iso19115_3/modules/module_geographic_extent.rb:69]: What purpose does this serve at the end here? is this somehow returning this variable?

File: lib/adiwg/mdtranslator/readers/iso19115_3/modules/module_reference_system.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ module Iso191153
             module ReferenceSystem
                @@vertRefSysPath = 'gex:verticalCRSId//mrs:MD_ReferenceSystem'
                @@refSysCodeXPath = 'mrs:referenceSystemType//mrs:MD_ReferenceSystemTypeCode'
-               @@refSysIdXPath = './/gmd:referenceSystemIdentifier'
+               @@refSysIdXPath = 'mcc:referenceSystemIdentifier'
```
Comment (jbrown-xentity) [inline lib/adiwg/mdtranslator/readers/iso19115_3/modules/module_reference_system.rb:14]: Why the change to mcc here? What is the difference?

File: lib/adiwg/mdtranslator/internal/module_utils.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+   def self.reconcile_hashes(hashA, hashB)
+      # merges hashes by value "truthyness"
+      # the "falsely" values we're looking to replace
+      # are nil, false, and empty {} & []
+      # there shouldn't be any situation with 2 truthy values
+      hashA.merge(hashB) do |_, oldval, newval|
+         (oldval.respond_to?(:empty?) && oldval.empty?) || !oldval ? newval : oldval
+      end
+   end
+end
```
Comment (jbrown-xentity) [inline lib/adiwg/mdtranslator/internal/module_utils.rb:13]: Still not totally clear on what situation this would be used in. So we're examining 2 objects, and removing anything that's "falsy", and merging into one?

**Review by jbrown-xentity — APPROVED** [review]
Looks good, comments are just for clarification/my own edification

File: lib/adiwg/mdtranslator/readers/iso19115_3/modules/module_geographic_extent.rb (hunk 1 above)
Comment (rshewitt) [reply lib/adiwg/mdtranslator/readers/iso19115_3/modules/module_geographic_extent.rb:69]: yes, `geoExtents` will be the function return. ruby doesn't require the "return" statement. I did find a situation where explicitly calling "return" produced the correct result but that was probably coincidental

File: lib/adiwg/mdtranslator/internal/module_utils.rb (hunk 3 above)
Comment (rshewitt) [reply lib/adiwg/mdtranslator/internal/module_utils.rb:13]: yeah so i'm not stoked on this and I want to avoid having to use it ( or completely remove it ). the reason why I created it is because `Individual` and `Organization` modules use a [contact](https://github.com/GSA/mdTranslator/blob/9b46e132ee5fa1095285a9c10a4e187c97c528c3/lib/adiwg/mdtranslator/internal/internal_metadata_obj.rb#L302-L319) object to store information inside of. Both of these modules/elements can contain a `CI_Contact` element relying on the same `contact` internal object. In the case of `eMailList` in the `contact` internal object that needs to bubble up from the `address` which can exist inside a contact element. I was probably bombarded with these considerations and made this as a result. I'm gonna revisit this.

Comment (jbrown-xentity) [reply lib/adiwg/mdtranslator/internal/module_utils.rb:13]: I think the logic makes sense and it has tests so it clearly works... I think I would like it to throw or log an error if the unexpected happens, that is the object has multiple truthy values. What should happen then?
